pkg/api/api.proto

```diff
@@ -513,3 +513,63 @@ message OptionalBool {
 message OptionalFileMode {
   uint32 value = 1;
 }
+
+// CompoundFieldOwners tracks 'plugin ownership' of compound fields, typically
+// maps or slices.
+message CompoundFieldOwners {
+    map<string, string> owners = 1;
```

Comment:
Bringing over https://github.com/containerd/nri/pull/163#discussion_r2070947490

This needs a comment describing what is stored in the `map<string, string> owners`.  From reading through `owners.go`, it looks like:

* The key of the map is the key of the compound value (such as env var name, mount destination, annotation name)
* The value of the map is the name of the owning plugin